### PR TITLE
Admin merchants features

### PR DESCRIPTION
**Pull request by Selena730**
User Stories 27,28 and 29 done.
User story 27: 
- A new column(status) was added to merchants table with a default value of 0 -enum used 
- A method was added to merchant model and updated the update method in controller
User story 28: 
- Groups merchants by status 
- No new methods needed
User story 29: 
- New html file was created (merchants/new.html.erb) 
- New spec file was added (merchants/new_spec.rb)

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -13,18 +17,43 @@ def edit
 
    def update
       @merchant = Merchant.find(params[:id])
-      if @merchant.update(merchant_params)
+      
+      if params[:update_status] == "disable"
+         @merchant.update!(status: :disabled)
+         redirect_to admin_merchants_path
+
+      elsif @merchant.update(merchant_params)
```
Comment (IgorrMagalhaess) [inline app/controllers/admin/merchants_controller.rb:25]: I love this if statement that checks for params update status!! Very organized!

File: app/controllers/admin/merchants_controller.rb [hunk 2]
```diff
@@ -13,18 +17,43 @@ def edit
...
          redirect_to admin_merchant_path(@merchant.id)
          flash[:alert] = "Merchant Successfully Updated"
       else
          flash[:alert] = "Error: #{error_message(@merchant.errors)}"
```
Comment (IgorrMagalhaess) [inline app/controllers/admin/merchants_controller.rb:29]: Great sad path!!

File: app/models/merchant.rb [hunk 3]
```diff
@@ -12,4 +13,13 @@ def top_five_customers
       .order("count_transactions DESC")
       .limit(5)
    end
+
+
+   def change_status
```
Comment (IgorrMagalhaess) [inline app/models/merchant.rb:18]: Do you still need this method?

File: spec/features/admin/merchants/index_spec.rb [hunk 4]
```diff
@@ -99,5 +102,62 @@
...
+            expect(page).to have_content(@brown_merchant.name)
+         end
+
+         within "#disabled_merchants" do 
+            expect(page).to have_content(@black_merchant.name)
+         end
+      end
+
+      # User Story 29 
+      it 'shows a link to create new merchant' do
```
Comment (IgorrMagalhaess) [inline spec/features/admin/merchants/index_spec.rb:147]: A good idea is to test the sad path here as well
`fill in("merchant[name]", with: " ")`
`expect(current_path) to eq(new_admin_path)`
`expect(page).to have_content("Error:")`

**Review by IgorrMagalhaess — APPROVED** [review]

File: app/models/merchant.rb (hunk 3 above)
Comment (Selena730) [reply app/models/merchant.rb:18]: nope!
I had it before adding some changes the controller update method
It’s gone now!

File: spec/features/admin/merchants/index_spec.rb (hunk 4 above)
Comment (Selena730) [reply spec/features/admin/merchants/index_spec.rb:147]: Thank you 
I will add it!

**Pull request merged**